hdfs-scheduler/src/main/java/org/apache/mesos/hdfs/scheduler/Reconciler.java

```diff
@@ -0,0 +1,110 @@
+package org.apache.mesos.hdfs.scheduler;
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.mesos.hdfs.state.IPersistentStateStore;
+import org.apache.mesos.Protos.TaskState;
+import org.apache.mesos.Protos.TaskStatus;
+import org.apache.mesos.Protos;
+import org.apache.mesos.SchedulerDriver;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.List;
+import java.util.Observable;
+import java.util.Observer;
+import java.util.Set;
+
+/**
+ * HDFS Mesos Framework Reconciler class implementation.
+ */
+public class Reconciler implements Observer {
+  private final Log log = LogFactory.getLog(HdfsScheduler.class);
+  private IPersistentStateStore store;
+  private Set<String> pendingTasks;
+
+  public Reconciler(IPersistentStateStore pStore) {
+    store = pStore;
+    pendingTasks = new HashSet<String>();
```

Comment:
HashSet<String> could be HashSet.   It is assigned to a Set<String> so <String> is implied... and enforced by the compiler.